src/forms/TimeInput.jsx

```diff
@@ -23,6 +54,66 @@ class TimeInput extends React.Component {
 		this.props.onChange && this.props.onChange(e);
 	}
 
+	/**
+	* @function onNumberChange
+	* @param e Event Object
+	* @description called when either the hour or minute input changes,
+	* 	in turn changes the state and updates the value of the inputs
+	*/
+	onNumberChange(e) {
+		const { value, id } = e.target;
+
+		id === 'hours' &&
+			this.setState(() => ({ hours: value }));
+
+		id === 'minutes' &&
+			this.setState(() => ({ minutes: value }));
+	}
+	/**
+	* @function onMeridianChange
+	* @param e Event Object
+	* @description called when the meridian <select> input changes,
+	* 	in turn changes the state and updates the value of the <select>
+	*/
+	onMeridianChange(e) {
+		const { value } = e.target;
+
+		this.setState(() => ({ meridian: value }));
+	}
+
+	/**
+	* @function onBlur
+	* @param e Event Object
+	* @description called when either the hour or minute input loses focus,
+	* 	and ensures the value entered is not out of the min/max range
+	*/
+	onBlur(e) {
+		const { value, min, max, id } = e.target;
+
+		if (value > max){
+			e.target.value = max;
+
+			id === 'hours' &&
+				this.setState(() => ({ hours: formatDigits(max) }));
+
+			id === 'minutes' &&
+				this.setState(() => ({ minutes: formatDigits(max) }));
+
+		} else if (value < parseInt(min)){
```

Comment:
`parseInt` should usually have a second arg of `10` (the 'radix') to avoid being parsed with a non-decimal base